src/main/java/frc/team1816/robot/commands/GamepadDriveCommand.java

```diff
@@ -0,0 +1,49 @@
+package frc.team1816.robot.commands;
+
+import com.edinarobotics.utils.gamepad.Gamepad;
+import edu.wpi.first.wpilibj.command.Command;
+import frc.team1816.robot.Components;
+import frc.team1816.robot.subsystems.Drivetrain;
+
+import static com.edinarobotics.utils.math.Math1816.coerceValue;
+
+public class GamepadDriveCommand extends Command {
+    private Drivetrain drivetrain;
+    private Gamepad gamepadDriver;
+
+    public GamepadDriveCommand(Gamepad gamepadDriver){
+        super("gamepaddrivecommand");
+        this.drivetrain = Components.getInstance().drivetrain;
+        this.gamepadDriver = gamepadDriver;
+        requires(drivetrain);
+    }
+
+    @Override
+    protected void initialize() {
+    }
+
+    @Override
+    protected void execute() {
+        double speed = gamepadDriver.getLeftY();
```

Comment:
Same comment here about using Controls.getInstance().gamepadDriver.

However one thing I'm noticing from looking at this code is that there is no one place to go where you can see what all the gamepad buttons and sticks are doing.  You have to look all over the code.

If you want to use better Object-Oriented code then the Controls class should really own the behavior or the gamepad.  For instance you could add a getSpeed() method to Controls that returns gamepadDriver.getLeftY().

Then this code doesn't have to know anything about how the controls are setup:
```
  double speed = Controls.getInstance().getSpeed();
  double turn = Controls.getInstance().getTurn();
```
again copying @andrewfhou